docs/resources/oxide_silo.md

```diff
@@ -0,0 +1,100 @@
+---
+page_title: "oxide_silo Resource - terraform-provider-oxide"
+---
+
+# oxide_silo (Resource)
+
+This resource manages the creation of an Oxide silo.
+
+-> This resource requires Terraform version v1.11 or later due to the
+`tls_certificates` attribute being a
+[write-only attribute](https://developer.hashicorp.com/terraform/plugin/framework/resources/write-only-arguments).
+
+-> Only the `quotas` attribute supports in-place modification. Changes to other
+attributes will result in the silo being destroyed and created anew.
+
+## Example Usage
+
+```hcl
+resource "oxide_silo" "example" {
+  name             = "showcase"
+  description      = "Demo and event silo."
+  admin_group_name = "showcase_admin"
+  identity_mode    = "saml_jit"
+  discoverable     = true
+  mapped_fleet_roles = {
+    admin  = ["admin", "collaborator"]
+    viewer = ["viewer"]
+  }
+  quotas = {
+    cpus    = 64
+    memory  = 137438953472 # 128 GiB
+    storage = 549755813888 # 512 GiB
+  }
+  tls_certificates = [
+    {
+      name        = "wildcard_cert"
+      description = "Wildcard cert for *.sys.oxide.example.com."
+      cert        = file("cert.pem")
+      key         = file("key.pem")
+      service     = "external_api"
+    },
+  ]
+}
+```
+
+## Schema
+
+### Required
+
+- `name` (String) Name of the Oxide silo.
+- `description` (String) Description for the Oxide silo.
+- `quotas` (Set of Object) Limits the amount of provisionable CPU, memory, and storage in the silo. (See [below for nested schema](#nestedatt--quotas).)
+- `tls_certificates` (String, Write-only) Initial TLS certificates to be used for the new silo's console and API endpoints. This attribute is a [write-only attribute](https://developer.hashicorp.com/terraform/plugin/framework/resources/write-only-arguments) and can only be modified by updating its configuration and [replacing the resource](https://developer.hashicorp.com/terraform/cli/state/taint). (See [below for nested schema](#nestedatt--tls).)
```

Comment:
It appears we're not requiring a replace on the schema https://github.com/oxidecomputer/terraform-provider-oxide/pull/425/files#diff-1ea56842b20e164d37586563e1ee70d692dbc252771685f2072ef22bc4706682R153-R160.

Also, perhaps we don't need to specify that this attribute requires replace? There are other attributes that require replace and we don't mention it in the docs (e.g. `name`).

Reply:
We can't require a replace on this since it's a write-only attribute and the API does not return the `tls_certificates` back to us. If we required a replace we'd always have a diff since the configuration would not match what's returned from `Read`.

Reply:
Gotcha, could we rephrase `and can only be modified by updating its configuration and [replacing the resource](https://developer.hashicorp.com/terraform/cli/state/taint).` then? I think I was a bit confused by this sentence in particular.